module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,61 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+void printMap(const bitmapArr& bitmap) {
+
+    for (const auto& arr : bitmap) {
+        for (auto el : arr) {
+            if (iscntrl(el))
+                std::cout << " ";
+            else
+                std::cout << el;
+        }
+        std::cout << "\n";
+    }
+}
+
+grayScaleVec compressGrayscale(const bitmapArr& grayScale) {
+
+    grayScaleVec compressed;
+    compressed.reserve(width * height);
+    uint8_t count = 0;
+
+    for (const auto& row : grayScale) {
+        auto prev = row.front();
+        for (auto column : row) {
+            if (prev == column) {
+                count++;
+                prev = column;
+                continue;
+            }
+            compressed.push_back({prev, count});
+            prev = column;
+            count = 1;
+        }
+        compressed.push_back({prev, count});
+        count = 0;
+    }
```

Comment:
You can try reconfigure this to use only one push_back, it is not so hard

Reply:
Also, when you do this, You will see that it helps you with refactoring in STL#2. Because when you only once use push_back, you can easily replace it with algorithm.